plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -100,71 +126,74 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
-         fc_ilog(_log, "incoming connection");
-         socket_stream = std::make_unique<ws::stream<tcp::socket>>(std::move(socket));
-         socket_stream->binary(true);
-         socket_stream->next_layer().set_option(boost::asio::ip::tcp::no_delay(true));
-         socket_stream->next_layer().set_option(boost::asio::socket_base::send_buffer_size(1024 * 1024));
-         socket_stream->next_layer().set_option(boost::asio::socket_base::receive_buffer_size(1024 * 1024));
-         socket_stream->async_accept([self = shared_from_this()](boost::system::error_code ec) {
+      session(std::shared_ptr<state_history_plugin_impl> plugin, SocketType socket)
+          : plugin(std::move(plugin)), socket_stream(std::move(socket)) {}
+
+      void start() {
+         ilog("incoming connection");
+         socket_stream.auto_fragment(false);
+         socket_stream.binary(true);
+         if constexpr (std::is_same_v<SocketType, tcp::socket>) {
+            socket_stream.next_layer().set_option(boost::asio::ip::tcp::no_delay(true));
+         }
+         socket_stream.next_layer().set_option(boost::asio::socket_base::send_buffer_size(1024 * 1024));
+         socket_stream.next_layer().set_option(boost::asio::socket_base::receive_buffer_size(1024 * 1024));
+        
+         socket_stream.async_accept([self = this->shared_from_this()](boost::system::error_code ec) {
             self->callback(ec, "async_accept", [self] {
                self->start_read();
                self->send(state_history_plugin_abi);
             });
          });
       }
+     
 
       void start_read() {
          auto in_buffer = std::make_shared<boost::beast::flat_buffer>();
-         socket_stream->async_read(
-             *in_buffer, [self = shared_from_this(), in_buffer](boost::system::error_code ec, size_t) {
+         socket_stream.async_read(
+             *in_buffer, [self = this->shared_from_this(), in_buffer](boost::system::error_code ec, size_t) {
                 self->callback(ec, "async_read", [self, in_buffer] {
                    auto d = boost::asio::buffer_cast<char const*>(boost::beast::buffers_front(in_buffer->data()));
                    auto s = boost::asio::buffer_size(in_buffer->data());
                    fc::datastream<const char*> ds(d, s);
                    state_request               req;
                    fc::raw::unpack(ds, req);
-                   std::visit(*self, req);
+                   app().post(priority::medium, [self, req = std::move(req)]() mutable { std::visit(*self, req); });
                    self->start_read();
                 });
              });
       }
 
-      void send(const char* s) {
-         send_queue.push_back({s, s + strlen(s)});
-         send();
-      }
-
-      template <typename T>
-      void send(T obj) {
-         send_queue.push_back(fc::raw::pack(state_result{std::move(obj)}));
-         send();
-      }
+      
 
-      void send() {
+       void send() {
```

Comment:
Stray whitespace.  Unnecessary whitespace changes should be avoided, and the majority of the codebase uses three spaces per block indent.

Reply:
Thank you, fixed.